docs/Coding/mapgen.rst

```diff
@@ -0,0 +1,569 @@
+map generators. ``wetness`` also sets the amount of rivers, ranging from 3% to
+11% of all land including mountains.
+
+The most complicated formulae are for swamp and desert, as they combine
+``wetness`` and ``temperature``. Swamps do not appear at all when the world is
+dry and cold, but their fraction increases with both ``wetness`` and
+``temperature`` --- slightly faster with the former. Deserts may also not appear
+if the world is very cold and wet. Their fraction increases rather quickly with
+``temperature`` and is mildly reduced by ``wetness``.
+
+.. _mapgen-height-generators:
+
+Height-Based Generators
+-----------------------
+
+Three of the available algorithms start by building a height map for the whole
+world: :ref:`mapgen-random` (``RANDOM``), :ref:`mapgen-fractal` (``FRACTAL``),
+and :ref:`Fracture Map <mapgen-fracture>` (``FRACTURE``).
+This height map is then used to assign tiles to continents or seas, and to
+distribute terrains on the map. The algorithm used for this is shared between
+the generators and is described in :ref:`mapgen-terrain-assignment`.
+
+.. _mapgen-random:
+
+Fully Random Height
+^^^^^^^^^^^^^^^^^^^
+
+This generator is extremely simple: it builds a completely random height map and
+smoothes it out.
+Terrain is then assigned to tiles based on their height and temperature as
+described in the :ref:`terrain assignment <mapgen-terrain-assignment>` section
+below, and :ref:`rivers are added on top <mapgen-height-rivers>`.
+
+.. _mapgen-fractal:
+
+Pseudo-Fractal Height
+^^^^^^^^^^^^^^^^^^^^^
+
+This generator works by dividing the map in blocks (five plus the number of
+player islands to be created) and assigning a random height to their
+corners. Each block is then processed recursively, cutting it equally in four
+blocks. The height at the corners of the smaller blocks are computed by
+averaging the heights at the corners of the large blocks and adding a decreasing
+amount of noise. This process is repeated until blocks are one tile wide or
+less.
+
+At the end of the generation, more random noise is added on top of the generated
+height map to give more variety on large maps, with warmer tiles generally
+getting more noise. The generated height map is then used to assign terrains as
+:ref:`described below <mapgen-terrain-assignment>`.
+
+.. _mapgen-fracture:
+
+Fracture Map
+^^^^^^^^^^^^
+
+The ``FRACTURE`` generator starts from points distributed randomly on the map
+and grows them until they meet their neighbors. Each point is given a random
+height, which is shared by all connected tiles. After adding some randomness
+on top, the lowest areas are flooded and only the highest ones remain as
+islands. Interesting structures with many small islands are often created in one
+or two areas generated right next to the sea level.
+
+Hills and mountains are added at the boundaries between areas in a crude mimic
+of plate tectonics; this results in large mountain ranges inland and hills all
```

Comment:
Style guide would simply keep these as two sentences

Reply:
I had forgotten how deeply English speakers hate nuance in punctuation :sweat_smile: